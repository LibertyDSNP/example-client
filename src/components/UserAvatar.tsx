import React from "react";
import { Avatar } from "antd";
import * as blockies from "blockies-ts";
import { User } from "../utilities/types";
import { ProfileQuery } from "../services/content";
import { userIdentification } from "../services/utilities";

const avatarSizeOptions = new Map([
  ["small", 32],
  ["medium", 50],
  ["large", 100],
]);

interface UserAvatarProps {
  user: User | undefined;
  avatarSize: string;
}

const UserAvatar = ({ user, avatarSize }: UserAvatarProps): JSX.Element => {
  const { data: profile } = ProfileQuery(user);

  const iconURL =
    profile?.icon?.[0]?.href ||
    blockies.create({ seed: user?.fromId || "unknown" }).toDataURL();

  const name = userIdentification(user, profile);

  return (
    <Avatar
      className="UserAvatar"
      alt={name}
      src={iconURL}
      size={avatarSizeOptions.get(avatarSize)}
      style={{ minWidth: avatarSizeOptions.get(avatarSize) }}
    />
  );
};

export default UserAvatar;
