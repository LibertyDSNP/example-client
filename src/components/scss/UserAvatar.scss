.UserAvatar__active {
  border: 2px solid $green;
}
