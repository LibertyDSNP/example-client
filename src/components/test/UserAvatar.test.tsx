import React from "react";
import { shallow } from "enzyme";
import UserAvatar from "../UserAvatar";
import { getPrefabProfile } from "../../test/testProfiles";

describe("UserAvatar", () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });
  it("renders without crashing", () => {
    const component = shallow(
      <UserAvatar profile={getPrefabProfile(0)} avatarSize="large" />
    );
    expect(component).toMatchSnapshot();
  });
});
