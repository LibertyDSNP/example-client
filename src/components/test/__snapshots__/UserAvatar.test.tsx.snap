// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatar renders without crashing 1`] = `
<Avatar
  alt="Monday January"
  shape="circle"
  size={100}
/>
`;
